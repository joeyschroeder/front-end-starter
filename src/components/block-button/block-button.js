import PropTypes from 'prop-types';
import React from 'react';
import styles from './block-button.style.scss';

export function BlockButton(props) {
  const { children, onClick } = props;

  return (
    <button className={styles.root} type="button" onClick={onClick}>
      {children}
    </button>
  );
}

BlockButton.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  onClick: PropTypes.func,
};

BlockButton.defaultProps = {
  children: null,
  onClick: null,
};
