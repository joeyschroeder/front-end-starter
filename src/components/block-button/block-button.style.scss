@import "../../styles/config/variables";
@import "../../styles/mixins/transition";

.root {
  @include transition(#{"background, transform"});

  background: $color-primary;
  border-radius: $border-radius;
  box-sizing: border-box;
  color: $color-white;
  display: block;
  font-weight: $font-weight-bold;
  margin-bottom: calc($spacer / 2);
  padding: calc($spacer / 2);
  text-transform: uppercase;
  width: 100%;

  &:active {
    transform: scale(0.9);
  }

  &:hover {
    background: $color-primary-dark;
    color: $color-white;
  }
}
