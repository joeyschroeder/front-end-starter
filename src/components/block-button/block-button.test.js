import { BlockButton } from './block-button';
import React from 'react';
import { create } from 'react-test-renderer';

describe('<BlockButton />', () => {
  it('should render correctly', () => {
    const component = create(<BlockButton />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
