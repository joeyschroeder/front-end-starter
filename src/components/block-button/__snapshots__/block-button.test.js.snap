// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockButton /> should render correctly 1`] = `
<button
  className="root"
  onClick={null}
  type="button"
/>
`;
